docs/rfc/0000-radicle_daemon.md

```diff
@@ -0,0 +1,303 @@
+but naturally, more will be added as the project evolves.
+
+* Gossip
+* Identity
+  * read
+  * write
+* Profile
+* Replication
+* Tracking
+
+We will sketch these capabilities here, but this may not reflect the final
+definition that will be found in the implementation.
+
+### Identity
+
+The identity capabilities define the create, read, and update methods
+for interacting with Radicle's family of `Identity` types. See
+[spec/identities][id] for more details.
+
+```rust
+pub trait read::Identity<I> {
+	type Error;
+	
+	fn get<R>(&self, urn: Urn<R>) -> Result<Option<I>, Self::Error>;
+	fn list<R>(&self) -> Result<impl Iterator<Item = Result<I, Self::Error>, Self::Error>;
+}
+
+pub trait write::Identity<I> {
+	type Error;
+	
+	fn create<P, D>(&self, payload: P, delegations: D) -> Result<I, Self::Error>;
+	fn update<P, D>(&self, urn: Urn<R>, payload: P, delegations: D) -> Result<I, Self::Error>;
+	fn merge<R>(&self, urn: Urn<R>, from: PeerId) -> Result<I, Self::Error>;
+}
+
+pub trait read::Rad<I> {
+	type Error;
+	type Person;
+	type Signatures;
+	
+	/// rad/self
+	fn whoami(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Person>, Self::Error>;
+	
+	/// rad/signed_refs
+	fn signatures(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Signatures>, Self::Error>;
+	
+	/// rad/ids/*
+	fn delegates(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<impl Iterator<Item = Result<Self::Person, Self::Error>>, Self::Error>;
+	
+	/// rad/ids/<id>
+	fn delegate(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Person>, Self::Error>;
+}
+```
+
+The `I` parameters signify that the traits are open to many
+identities, and for now there would be specific implementations for
+`Person` and `Project`. We leave any domain specific types open using
+associated types on the trait, e.g. `Rad::Person = Person` will be
+associated type for a `Project`'s `rad/self`.
+
+### Tracking
+
+As well as the family of `Identity` capabilities, we will also want
+methods for the tracking graph of an identity, sketched below:
+
+```rust
+pub trait Tracking {
+	type Error;
+	type Tracked;
+	
+	fn track<R>(&self, urn: Urn<R>) -> Result<bool, Self::Error>;
+	fn untrack<R>(&self, urn: Urn<R>) -> Result<bool, Self::Error>;
+	fn tracked<R>(&self, urn: Urn<R>) -> Result<Tracked, Self::Error>;
+}
+```
+
+### Profile
+
+The profile capability defines how a person can create a profile, read
+the current profile, and switch between existing profiles. 
+
+```rust
+pub trait Profile {
+	type Error;
+	type LocalIdentity;
+	
+	fn create(&self, whomai: Self::LocalIdentity) -> Result<(), Self::Error>;
+	fn update(&self, whoami: Self::LocalIdentity) -> Result<(), Self::Error>;
+	fn current(&self) -> Result<Option<Self::LocalIdentity>, Self::Error>;
+	fn switch(&self, other: Self) -> Result<Self, Self::Error>;
+}
+```
+
+There is an open question of how `Profile` will interact with a
```

Comment:
Does “profile” refer to the mechanism to switch the state and config directories? If so, I have some reservations:

If the idea is that only one `daemon` process is run per logged-in user, every request by a client will need to assert that it is operating on the profile it expects — if it is possible to change the profile programmatically, another client could change it in-between requests. It also needs to prove with every request that it owns the private key associated with the profile’s peer id (although that is the case anyway). The only way I could see this working is if the daemon supports running multiple profiles concurrently, and have start/stop/status endpoints for each one. That does seem to broaden the scope considerably, so I’d recommend to punt on this for now, and require using the system process manager to switch profiles.

I’m also not excited about pretending that the mapping between profile and identity is 1:1

Reply:
> Does “profile” refer to the mechanism to switch the state and config directories?

Ya, that's what I was going for. Fair points, well made :) I'll strip back the `switch` and leave `create` and `current`. I'll also leave out `update` since that could allow a pseudo-switch, which seems like a bad idea in hindsight.